contrib/mathjax-compat/mathjax-compat.js

```diff
@@ -0,0 +1,28 @@
+/* global katex */
+
+function replaceMathJaxInline(node) {
+    const katexinline = document.createElement("span");
+    katexinline.setAttribute("class", "inline-equation");
+    katexinline.innerHTML = katex.renderToString(node.text);
+    node.parentNode.replaceChild(katexinline, node);
+}
+
+function replaceMathJaxDisplay(node) {
+    const katexinline = document.createElement("div");
+    katexinline.setAttribute("class", "equation");
+    katexinline.innerHTML = "\\displaystyle " + katex.renderToString(node.text);
+    node.parentNode.replaceChild(katexinline, node);
+}
+
+{
+    const l = document.scripts.length;
+    for (let i = 0; i < l; i++) {
+        const s = document.scripts[i];
+        if (s.type === "math/tex") {
+            replaceMathJaxInline(s);
+        }
+        if (s.type === "math/tex; display") {
```

Comment:
Looks like MathJax does two checks. [The first check](https://github.com/mathjax/MathJax/blob/6273842a9746731b9ecca0de18ec9fd50a36df99/unpacked/MathJax.js#L2022) removes `/ *;.*/` (i.e. whitespace and then a semicolon and everything after the semicolon) and then checks for "math/tex".

Then, [the second check](https://github.com/mathjax/MathJax/blob/master/unpacked/jax/input/TeX/jax.js#L2168) looks for an instance of `/(;|\s|\n)mode\s*=\s*display(;|\s|\n|$)/` to decide if it should be in display mode. Those could probably be combined into a single check, I think it's just done in two parts because of how MathJax is structured.